src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -144,6 +145,10 @@ public String actualPathForPublicPath(String publicPath) {
             String lectureId = publicPath.replace(filename, "").replace("/api/files/attachments/lecture/", "");
             return Constants.LECTURE_ATTACHMENT_FILEPATH + lectureId + filename;
         }
+        if (publicPath.contains("files/file-upload-exercises")) {
+            String[] subpath = publicPath.replace(filename, "").replace("/api/files/file-upload-exercises/", "").split("/");
```

Comment:
This is bad code. I think you try to parse some ids here, then please also use variable names that include the ids like above `lectureId`. Also make sure that this code cannot fail by checking if these elements are actually available

Reply:
Added explicit number check and used new variables